src/coreclr/nativeaot/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs

```diff
@@ -262,9 +262,44 @@ private static unsafe void RehydrateData(IntPtr dehydratedData, int length)
                 switch (command)
                 {
                     case DehydratedDataCommand.Copy:
-                        // TODO: can we do any kind of memcpy here?
-                        for (; payload > 0; payload--)
-                            *pDest++ = *pCurrent++;
+                        Debug.Assert(payload != 0);
+                        if (payload < 4)
+                        {
+                            *pDest = *pCurrent;
+                            if (payload > 1)
+                                *(short*)(pDest + payload - 2) = *(short*)(pCurrent + payload - 2);
+                        }
+                        else if (payload < 8)
+                        {
+                            *(int*)pDest = *(int*)pCurrent;
+                            *(int*)(pDest + payload - 4) = *(int*)(pCurrent + payload - 4);
```

Comment:
Can we get in trouble with alignment here? Sometimes misaligned accesses are a lot slower. Especially when crossing cache lines.

Reply:
I thought about that, but this is basically `Buffer.MemoryCopy`. If it didn't matter for the general purpose API, it probably doesn't matter here either.